src/featbin/wav-reverberate.cc

```diff
@@ -0,0 +1,255 @@
+        KALDI_WARN << "options for --rir-channel and --noise-channel will be ignored"
+                  << "when --multi-channel-output is true.";
+    }
+
+    std::string input_wave_file = po.GetArg(1);
+    std::string rir_file = po.GetArg(2);
+    std::string output_wave_file = po.GetArg(3);
+
+    WaveData input_wave;
+    {
+      Input ki(input_wave_file);
+      input_wave.Read(ki.Stream());
+    }
+
+    const Matrix<BaseFloat> &input_matrix = input_wave.Data();
+    BaseFloat samp_freq_input = input_wave.SampFreq();
+    int32 num_samp_input = input_matrix.NumCols(),  // number of samples in the input
+          num_input_channel = input_matrix.NumRows();  // number of channels in the input
+    KALDI_VLOG(1) << "sampling frequency of input: " << samp_freq_input
+                  << " #samples: " << num_samp_input
+                  << " #channel: " << num_input_channel;
+    KALDI_ASSERT(input_channel < num_input_channel);
+
+    WaveData rir_wave;
+    {
+      Input ki(rir_file);
+      rir_wave.Read(ki.Stream());
+    }
+    const Matrix<BaseFloat> &rir_matrix = rir_wave.Data();
+    BaseFloat samp_freq_rir = rir_wave.SampFreq();
+    int32 num_samp_rir = rir_matrix.NumCols(),
+          num_rir_channel = rir_matrix.NumRows();
+    KALDI_VLOG(1) << "sampling frequency of rir: " << samp_freq_rir
+                  << " #samples: " << num_samp_rir
+                  << " #channel: " << num_rir_channel;
+    if (!multi_channel_output) {
+      KALDI_ASSERT(rir_channel < num_rir_channel);
+    }
+
+    Matrix<BaseFloat> noise_matrix;
+    if (!noise_file.empty()) {
+      WaveData noise_wave;
+      {
+        Input ki(noise_file);
+        noise_wave.Read(ki.Stream());
+      }
+      noise_matrix = noise_wave.Data();
+      BaseFloat samp_freq_noise = noise_wave.SampFreq();
+      int32 num_samp_noise = noise_matrix.NumCols(),
+            num_noise_channel = noise_matrix.NumRows();
+      KALDI_VLOG(1) << "sampling frequency of noise: " << samp_freq_noise
+                    << " #samples: " << num_samp_noise
+                    << " #channel: " << num_noise_channel;
+      if (multi_channel_output) {
+        KALDI_ASSERT(num_rir_channel == num_noise_channel);
+      } else {
+        KALDI_ASSERT(noise_channel < num_noise_channel);
+      }
+    }
+
+    int32 num_output_channels = (multi_channel_output ? num_rir_channel : 1);
+    Matrix<BaseFloat> out_matrix(num_output_channels, num_samp_input);
+
+    for (int32 output_channel = 0; output_channel < num_output_channels; output_channel++) {
+      Vector<BaseFloat> input(num_samp_input);
+      input.CopyRowFromMat(input_matrix, input_channel);
+      float power_before_reverb = VecVec(input, input) / input.Dim();
+
+      int32 this_rir_channel = (multi_channel_output ? output_channel : rir_channel);
+      Vector<BaseFloat> rir(num_samp_rir);
+      rir.CopyRowFromMat(rir_matrix, this_rir_channel);
+      rir.Scale(1 / pow(2, 15));
```

Comment:
replace the 1, 2 and 15 with 1.0, 2.0 and 15.0 please.  better to use literal floats if you intend them to be evaluated in a floating-point context, as you do here.  1.0 / (1 << 15) would perhaps be preferable, though, as no dependency on -lmath.